### ISSUE TITLE
sync from onedriver ,Invalid API or resource

### ISSUE THREAD

**Issue by xiaowj** [laurent22/joplin#4181]
when sync from onedriver, joplin throw exception Error: Invalid API or resource

## Environment

Joplin version: Joplin 1.4.19 (prod, linux)
Platform: Ubnutu
OS specifics: 18.04

## Steps to reproduce

1. ckick sync

## Logfile
```
/tmp/.mount_JoplinHKiQZy/resources/app.asar/node_modules/@joplin/lib/Logger.js:165 11:47:13: Error: Invalid API or resource
    at OneDriveApi.oneDriveErrorResponseToError (/tmp/.mount_JoplinHKiQZy/resources/app.asar/node_modules/@joplin/lib/onedrive-api.js:130)
    at OneDriveApi.exec (/tmp/.mount_JoplinHKiQZy/resources/app.asar/node_modules/@joplin/lib/onedrive-api.js:276)
    at async OneDriveApi.execJson (/tmp/.mount_JoplinHKiQZy/resources/app.asar/node_modules/@joplin/lib/onedrive-api.js:346)
    at async FileApiDriverOneDrive.delta (/tmp/.mount_JoplinHKiQZy/resources/app.asar/node_modules/@joplin/lib/file-api-driver-onedrive.js:243)
    at async tryAndRepeat (/tmp/.mount_JoplinHKiQZy/resources/app.asar/node_modules/@joplin/lib/file-api.js:38)

```

**Comment by aaronxn** [comment]
I'm experiencing the same error. 

![Screenshot from 2020-12-07 08-33-44](https://user-images.githubusercontent.com/15862474/101370637-f8fbc000-3866-11eb-8e38-ff983f244e5e.png)
---
Joplin 1.4.19 (prod, linux)

Client ID: 9d3b4f5a84bf4fd5bc858d808a5e2a3c
Sync Version: 2
Profile Version: 34
Keychain Supported: No

Revision: de123ee (master)

**Comment by whikwon** [comment]
I'm also experiencing the same error.

**Comment by tonylaw7** [comment]
Also experiencing this issue with both Windows and MacOS joplin apps. Joplin iOS app seems to be stuck in a loop fetching items without stopping.

**Comment by Jon-Schneider** [comment]
New user as of today. I've been receiving this on my iPhone and multiple Macs all day.

**Comment by hu-jinwen** [comment]
me too

**Comment by cyberpunkrooster** [comment]
Same issue appearing on Ubuntu and iOS (latest versions)
Other observations:
- Creating a new note on Ubuntu and clicking Synchronise appears to successfully upload the note, but will still display the error.
- Wiping the iOS app, reinstalling and kicking off a sync successfully pulls all notes but will display the error again once everything is pulled.

**Comment by gagin** [comment]
Same here

**Comment by Hasenpfote** [comment]
I'm having the same problem when I'm syncing between Linux Mint and Android via OneDrive.
When this problem occurs, the OneDrive side is up to date, but the local side seems to not be reflected. (i.e. synchronization does not work.)

**Comment by kenkenspin** [comment]
Same here. Linux and iOS both showing the same error in sync via OneDrive too.
i remove .config/joplin-desktop/database.sql, once success sync, but synchronization does no work as showing Invalid API error.

**Comment by philc511** [comment]
Seeing same error on Windows, Joplin versions 1.4.18 and 1.2.6. Have re-installed and cleared out OneDrive Apps/Joplin, but the error recurs.
Also seeing it on latest iOS App too.

**Comment by koonmeng** [comment]
Same issue here, all my Windows and Android devices are continuously syncing with OneDrive. Seems to be syncing all my notes. After they complete it would start all over again during the next sync.

**Comment by FPuls** [comment]
My Mac seems to kinda sync it says completed and then shows the error. My phone just keeps counting and never stops

**Comment by jonath92** [comment]
Seems an upstream issue: https://github.com/OneDrive/onedrive-api-docs/issues/1408

**Comment by danfoxley** [comment]
> New user as of today. I've been receiving this on my iPhone and multiple Macs all day.

Same here new user.  Don't use DropBox, so OneDrive is preferred.

**Comment by DarkFuria** [comment]
Infinite fetch. Nothing useful on the console.

**Comment by whikwon** [comment]
Now it works correctly.

**Comment by cyberpunkrooster** [comment]
Still getting the error here across iOS and Ubuntu devices.

**Comment by xiaowj** [comment]
Now it works correctly.

**Comment by tonylaw7** [comment]
It's only working on iOS for me. Both MacOS and Windows clients are still showing the same error, can someone else confirm?

**Comment by kenkenspin** [comment]
fix it as link [OneDrive/onedrive-api-docs#1408](https://github.com/OneDrive/onedrive-api-docs/issues/1408).
I need reset config file, OK on both Linux and iOS. it remove(rename) .config/joplin-desktop/database.sql on Linux, remove iOS app, then reconfig for OneDrive.

**Comment by laurent22** [comment]
Re-opening as it's not fixed for everybody.
